fe/src/main/java/org/apache/doris/common/util/DynamicPartitionUtil.java

```diff
@@ -0,0 +1,251 @@
+            }
+            // dynamic partition enable default to true
+            if (Strings.isNullOrEmpty(enable)) {
+                enable = Boolean.TRUE.toString();
+                properties.put(DynamicPartitionProperty.ENABLE, enable);
+            }
+        }
+    }
+
+    public static Map<String, String> analyzeDynamicPartition(Database db, OlapTable olapTable,
+                                                              Map<String, String> properties) throws DdlException {
+        // properties should not be empty, check properties before call this function
+        Map<String, String> analyzedProperties = new HashMap<>();
+        if (properties.containsKey(DynamicPartitionProperty.TIME_UNIT)) {
+            String timeUnitValue = properties.get(DynamicPartitionProperty.TIME_UNIT);
+            checkTimeUnit(timeUnitValue);
+            properties.remove(DynamicPartitionProperty.TIME_UNIT);
+            analyzedProperties.put(DynamicPartitionProperty.TIME_UNIT, timeUnitValue);
+        }
+        if (properties.containsKey(DynamicPartitionProperty.PREFIX)) {
+            String prefixValue = properties.get(DynamicPartitionProperty.PREFIX);
+            checkPrefix(prefixValue);
+            properties.remove(DynamicPartitionProperty.PREFIX);
+            analyzedProperties.put(DynamicPartitionProperty.PREFIX, prefixValue);
+        }
+        if (properties.containsKey(DynamicPartitionProperty.END)) {
+            String endValue = properties.get(DynamicPartitionProperty.END);
+            checkEnd(endValue);
+            properties.remove(DynamicPartitionProperty.END);
+            analyzedProperties.put(DynamicPartitionProperty.END, endValue);
+        }
+        if (properties.containsKey(DynamicPartitionProperty.BUCKETS)) {
+            String bucketsValue = properties.get(DynamicPartitionProperty.BUCKETS);
+            checkBuckets(bucketsValue);
+            properties.remove(DynamicPartitionProperty.BUCKETS);
+            analyzedProperties.put(DynamicPartitionProperty.BUCKETS, bucketsValue);
+        }
+        if (properties.containsKey(DynamicPartitionProperty.ENABLE)) {
+            String enableValue = properties.get(DynamicPartitionProperty.ENABLE);
+            checkEnable(db, olapTable, enableValue);
+            properties.remove(DynamicPartitionProperty.ENABLE);
+            analyzedProperties.put(DynamicPartitionProperty.ENABLE, enableValue);
+        }
+        return analyzedProperties;
+    }
+
+    public static void checkAlterAllowed(OlapTable olapTable) throws DdlException {
+        TableProperty tableProperty = olapTable.getTableProperty();
+        if (tableProperty != null &&
+                tableProperty.getDynamicPartitionProperty().isExist() &&
+                Boolean.parseBoolean(tableProperty.getDynamicPartitionProperty().getEnable())) {
+            throw new DdlException("Cannot modify partition on a Dynamic Partition Table, set `dynamic_partition.enable` to false firstly.");
```

Comment:
Why not support drop partition in a dynamic partition table?
I think it will be OK to allow adding and droping partition to a dynamic partition table.
Dynamic partition only helps user to create or drop partition automatically, if partition has been created manually, dynamic partition scheduler just skip it and continue to handle other table.

Reply:
I think when user want to add or drop a partition to a dynamic partition table, they should kown what they are doing, and make sure this is a dynamic partition table.If user make sure he wants to adding or droping partition to a dynamic partition table, they can stop dynamic partition feature and I have give the tips in error message.

Reply:
It is OK that don't allow user to create partition for dynamic partition table. However I'm not very clear about the specific hazards of allowing users to create partitions. If the harm is not so serious, can we release this restriction?

Reply:
In my opinion, the harm is not so serious, I just want to make sure that user don't forget that he is adding or droping on a dynamic partition table, unless it is necessary.Release this restriction is ok for me.

Reply:
> It is OK that don't allow user to create partition for dynamic partition table. However I'm not very clear about the specific hazards of allowing users to create partitions. If the harm is not so serious, can we release this restriction?

Forbidding user to modify partition is just for simplify our logic. Otherwise there may be a large number of validation checks, concurrent conflict checks, and easy to cause errors.